server/src/main/java/org/opensearch/cluster/routing/RoutingNodes.java

```diff
@@ -368,26 +370,36 @@ public ShardRouting activePrimary(ShardId shardId) {
      * Returns one active replica shard for the given shard id or <code>null</code> if
      * no active replica is found.
      *
-     * Since replicas could possibly be on nodes with a older version of OpenSearch than
-     * the primary is, this will return replicas on the highest version of OpenSearch.
+     * Since replicas could possibly be on nodes with an older version of OpenSearch than
+     * the primary is, this will return replicas on the highest version of OpenSearch when
+     * document replication strategy is in use, and will return replicas on oldest version
+     * of OpenSearch when segment replication is enabled.
      *
      */
-    public ShardRouting activeReplicaWithHighestVersion(ShardId shardId) {
+    public ShardRouting activeReplicaBasedOnReplicationStrategy(ShardId shardId) {
```

Comment:
For readability, Rather than updating this method can we add a new `activeReplicaWithLowestVersion` and check the setting from  unassignPrimaryAndPromoteActiveReplicaIfExists?

```java
    private void unassignPrimaryAndPromoteActiveReplicaIfExists(
        ShardRouting failedShard,
        UnassignedInfo unassignedInfo,
        RoutingChangesObserver routingChangesObserver
    ) {
        assert failedShard.primary();
        final ShardRouting activeReplica;
        if (isSegmentRbeplicationEnabled) {
            activeReplica = activeReplicaWithHighestVersion(failedShard.shardId());
        } else {
            activeReplica = activeReplicaWithLowestVersion(failedShard.shardId());
        }
        if (activeReplica == null) {
            moveToUnassigned(failedShard, unassignedInfo);
        } else {
            movePrimaryToUnassignedAndDemoteToReplica(failedShard, unassignedInfo);
            promoteReplicaToPrimary(activeReplica, routingChangesObserver);
        }
    }
```